pyramid_swagger/tween.py

```diff
@@ -268,7 +268,10 @@ def files(self):
         return result
 
     def json(self, **kwargs):
-        return getattr(self.request, 'json_body', {})
+        if self.request.is_body_readable:
+            return getattr(self.request, 'json_body', {})
+        else:
+            return None
```

Comment:
Why are you using `None` here instead of the empty dictionary like above? It seems to me like `None` would be the better return value, but I think it should be consistent (and changing it might be backwards incompatible).

Reply:
I think that defaulting the json_body to empty dictionary is not correct as the body of a request could potentially be of `type: string`.

About consistency the main concern is that we use `None` as sentinel for missing parameter.
If we return something that is not None that value will eventually be validated and it could cause validation error for a parameter that has not been defined in the request


Anyway that was used to handle the case that `self.request` does not have `json_body` attribute. `self.request` comes from `WebOb` library (which has `json_body` defined by really old versions), so there should be no instances on which line 272 returns the default value